### PR TITLE
(MODULES-10413) Allow custom ntp user and daemon options 

### PR DESCRIPTION
**Pull request by david22swan**
Rebase of https://github.com/puppetlabs/puppetlabs-ntp/pull/489

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -13,6 +13,71 @@
...
+      $daemon_config = '/etc/sysconfig/ntp'
+      if $ntp::daemon_extra_opts {
+        file_line { 'Set NTPD daemon options':
+          ensure => present,
+          path   => $daemon_config,
+          line   => "OPTIONS='${ntp::daemon_extra_opts}'",
+          match  => '^OPTIONS\=',
+        }
+      }
+      # if $ntp::user {
```
Comment (sheenaajay) [inline manifests/config.pp]: @david22swan should we keep the commented block or remove them

Comment (david22swan) [reply manifests/config.pp]: Removed :)

File: REFERENCE.md [hunk 2]
```diff
@@ -82,6 +82,14 @@ Data type: `Optional[String]`
 Specifies an absolute or relative file path to an ERB template for the config file.
 Example value: 'ntp/ntp.conf.erb'. A validation error is thrown if both this **and** the `config_epp` parameter are specified.
 
+##### `daemon_extra_opts`
+
+Data type: `Optional[String]`
+
+Specifies any arguments to pass to ntp daemon. Default value: '-g'.
+Example value: '-g -i /var/lib/ntp' to enaible jaildir options.
```
Comment (sheenaajay) [inline REFERENCE.md]: small type 'enable'

File: REFERENCE.md [hunk 3]
```diff
@@ -443,6 +451,14 @@ Specifies the stratum the server should operate at when using the undisciplined
 This value should be set to no less than 10 if ntpd might be accessible outside your immediate, controlled network.
 Default value: 10.am udlc
 
+##### `user`
+
+Data type: `Optional[String]`
+
+Specifies user to run ntpd daemon. Default value: ntp.
+Usually set by default on Centos7 (/etc/systemd/system/multi-user.target.wants/ntpd.service) and ubuntu 18.04 (/usr/lib/ntp/ntp-systemd-wrapper)
+This is currently restricted to Redhat based systems of version 7 and aboce and Ubuntu 18.04.
```
Comment (sheenaajay) [inline REFERENCE.md]: small type 'above'

File: manifests/init.pp [hunk 4]
```diff
@@ -28,6 +28,11 @@
 #   Specifies an absolute or relative file path to an ERB template for the config file.
 #   Example value: 'ntp/ntp.conf.erb'. A validation error is thrown if both this **and** the `config_epp` parameter are specified.
 #
+# @param daemon_extra_opts
+#   Specifies any arguments to pass to ntp daemon. Default value: '-g'.
+#   Example value: '-g -i /var/lib/ntp' to enaible jaildir options.
```
Comment (sheenaajay) [inline manifests/init.pp]: small typo 'enable'

File: manifests/init.pp [hunk 5]
```diff
@@ -218,6 +223,11 @@
 #   This value should be set to no less than 10 if ntpd might be accessible outside your immediate, controlled network.
 #   Default value: 10.am udlc
 #
+# @param user
+#   Specifies user to run ntpd daemon. Default value: ntp.
+#   Usually set by default on Centos7 (/etc/systemd/system/multi-user.target.wants/ntpd.service) and ubuntu 18.04 (/usr/lib/ntp/ntp-systemd-wrapper)
+#   This is currently restricted to Redhat based systems of version 7 and aboce and Ubuntu 18.04.
```
Comment (sheenaajay) [inline manifests/init.pp]: small typo 'above'

Comment (david22swan) [reply manifests/init.pp]: fixed :)

File: manifests/init.pp (hunk 4 above)
Comment (david22swan) [reply manifests/init.pp]: fixed :)

File: REFERENCE.md (hunk 3 above)
Comment (david22swan) [reply REFERENCE.md]: fixed :)

File: REFERENCE.md (hunk 2 above)
Comment (david22swan) [reply REFERENCE.md]: fixed :)

**Comment by david22swan** [comment]
Screen Shot 2020-01-10 at 5 25 06 PM

**Pull request merged**